src/Worker/ReschedulingDecorator.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Neighborhoods\KojoWorkerDecoratorComponent\Worker;
+
+use DateTime;
+use Doctrine\DBAL\Connection;
+use LogicException;
+use Neighborhoods\KojoWorkerDecoratorComponent\WorkerInterface;
+use Neighborhoods\ThrowableDiagnosticComponent\ThrowableDiagnostic;
+use OutOfBoundsException;
+use Throwable;
+
+final class ReschedulingDecorator implements ReschedulingDecoratorInterface
+{
+    use DecoratorTrait;
+    use ThrowableDiagnostic\Builder\Factory\AwareTrait;
+
+    public const MAX_RESCHEDULE_DELAY_SECONDS = 86400;// (24 hours)
```

Comment:
The main reason is that we would expect a job to use a cron expression if it runs daily or less frequently.

I could add this into the readme.

Reply:
But if a user wants to and has a valid reason to, is there a technical reason why a user should be prevented from doing so? Shouldn't this set of components be providing flexible and customizable tools for a variety of use-cases and not creating restrictions on how a user should be doing things? From a cron-expression perspective, there's not really any difference between something being weekly, daily, hourly, or every minute, so deciding that clients should exclusively use cron-expressions above one of those thresholds feels arbitrarily restrictive from a user perspective.

Reply:
Limit has been completely removed.

Reply:
Though I did just think of something where someone may want to specify a maximum limit (might be a new decorator) is retrying with an exponential backoff. We do something like that in Listing Service in a few places.

Reply:
I think that would go into the Exception Handling Decorator. Currently the exception handling decorator just delays by a constant time in case of a transient exception. I guess you could have different delay strategies and specify which one you want using DI.
That's out of scope for this ticket. Please use the SERCOMP project if you want to add it to the backlog